pkgs/by-name/ke/keep-sorted/package.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
+  pname = "keep-sorted";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "keep-sorted";
+    rev = "v${version}";
+    hash = "sha256-bCV0XcwgyFTORl/RF1BS7vsM8DmU0Wox3OIEuZBrwSs=";
+  };
+
+  vendorHash = "sha256-yaeqfMAJbQdrqZ0uco6Y5T8vnfjlBJY4IQuGzZg3Ubw=";
+
+  CGO_ENABLED = "0";
```

Comment:
```suggestion
  env.CGO_ENABLED = "0";
```

Reply:
Why? Couldn't find this in any other package:

```
❯ rg -l 'CGO_ENABLED' | wc -l
96
❯ rg -l 'env.CGO_ENABLED' | wc -l
0
```

Reply:
It's a new thing and doesn't pollute the attribute set.

Reply:
Tried it, doesn't work:

```
error: The ‘env’ attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping: CGO_ENABLED
```

Reply:
I also couldn't find any documentation of this in the [nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/). So it might be a good idea to document this first.

Reply:
> Tried it, doesn't work:
> 
> ```
> error: The ‘env’ attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping: CGO_ENABLED
> ```

Because the go builders set it on the derivation. That should be fixed.

Reply:
It is set on the derivation so it can default to `go.CGO_ENABLED` I guesss:
https://github.com/NixOS/nixpkgs/blob/98426e81f4e2b8446c9ccb1fca584adb2c6a804b/pkgs/build-support/go/module.nix#L35

@mweinelt is the env thing documented anywhere?

Reply:
@Artturin might know, I'm not sure and grepping for it requires playing in hard mode.